pkgs/development/libraries/portmidi/default.nix

```diff
@@ -44,14 +46,38 @@ stdenv.mkDerivation rec {
     ln -s libportmidi.so "$out/lib/libporttime.so"
   '';
 
-  buildInputs = [ unzip cmake /*jdk*/ alsaLib ];
+  preConfigure = optional stdenv.isDarwin (''
+    substituteInPlace ./CMakeLists.txt --replace "i386 ppc x86_64" "x86_64"
+    substituteInPlace ./pm_common/CMakeLists.txt --replace "/Developer/SDKs/MacOSX10.5.sdk" \
+      "${apple_sdk.sdk}"
+
+  '' +
+  concatMapStrings
+  (path:
+  ''
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreAudio.framework" \
+      "${CoreAudio}/Library/Frameworks/CoreAudio.framework"
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreFoundation.framework" \
+      "${CF}/Library/Frameworks/CoreFoundation.framework"
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreMIDI.framework" \
+      "${CoreMIDI}/Library/Frameworks/CoreMIDI.framework"
+    substituteInPlace ./${path}/CMakeLists.txt --replace "CoreServices.framework" \
+      "${CoreServices}/Library/Frameworks/CoreServices.framework"
```

Comment:
To clarify, I think something like this should make most of the patch file redundant.

```
substituteInPlace pm_dylib/CMakeLists.txt \
    --replace '${FRAMEWORK_PATH}/CoreAudio.framework' "${CoreAudio}/Library/Frameworks/CoreAudio.framework"
...
```